lib/ruby/lib/alces/stack/boot/run.rb

```diff
@@ -34,113 +34,118 @@ class Run
-          
-          # Sets dynamic variables in the kickstart options
-          @kickstart_options[:nodename] = json_new[:nodename] if json_new.key?("nodename".to_sym)
-          raise "No node specified for appending kickstart file" if !@kickstart_options.key?("nodename".to_sym)
-          @kickstart_options[:save_append] = "." << @kickstart_options[:nodename]
-          @kickstart_options[:json] = json_new.to_json
-          @kickstart_files << Alces::Stack::Kickstart.run!(@kickstart, @kickstart_options)
-          puts "\n" if @dry_run_flag
+        def add_kickstart(parameters={})
+          ks_file = @ks_finder.filename_diff_ext("ks")
+          ks_file << "." << parameters[:nodename]
+          parameters[:kickstart] = ks_file
         end
 
-        def set_kickstart_template_parameter
-          @kickstart = Alces::Stack::Templater::Finder.new("#{ENV['alces_BASE']}/etc/templates/kickstart/").find(@kickstart)
-          @kickstart_name = @kickstart.scan(/\.?\w+\.?\w*\Z/)
-          raise "Could not determine kickstart file name: #{@kickstart}" if @kickstart_name.size != 1
-          @kickstart_name = @kickstart_name[0].scan(/\.?\w+/)[0] << ".ks"
-          @template_parameters[:kickstart] = "#{@kickstart_name}.<%= nodename %>" 
-        
-          @kickstart_options = {
+        def run_kickstart
+          kickstart_options = {
             group: false,
-            dry_run_flag:  @dry_run_flag,
-            ran_from_boot: true
+            dry_run_flag: @dry_run_flag,
+            ran_from_boot: true,
+            json: @json
           }
-          @kickstart_options[:nodename] = @template_parameters[:nodename] if @template_parameters.key?("nodename".to_sym)
-          kickstart_lambda = -> (json) {run_kickstart(json)}
-          @kickstart_files = Array.new
-          Alces::Stack::Iterator.new(@group, kickstart_lambda, @json)
+          kickstart_options[:nodename] = @template_parameters[:nodename] if @template_parameters.key?("nodename".to_sym)
+          kickstart_lambda = -> (hash) {
+            hash[:save_append] = hash[:nodename]
+            return Alces::Stack::Kickstart::Run.new(@kickstart_template, hash).run!
+          }
+          kickstart_files = Alces::Stack::Iterator.run(@group, kickstart_lambda, kickstart_options)
+          @to_delete_dry_run.push(*kickstart_files) if @dry_run_flag
+          @to_delete.push(*kickstart_files) if !@dry_run_flag
         end
 
         def kickstart_teardown
           # Deletes old signal files
           delete_lambda = -> (options) { `rm -f /var/lib/metalware/cache/metalwarebooter.#{options[:nodename]}` }
           Alces::Stack::Iterator.run(@group, delete_lambda, {nodename: @template_parameters[:nodename]})
-          # 
+
+          # Gets the kick start template filename
+          ks_finder = Alces::Stack::Templater::Finder.new("#{ENV['alces_BASE']}/etc/templates/kickstart/")
+          ks_finder.template = @kickstart_template
```

Comment:
When I was originally designing `Finder` I wasn't sure if the same object was going to need to find multiple templates in the future. So that was major design consideration.

If that is not going to be the case for the production code (ex. tests) I'll make a change